src/blob/blob_client_wrapper.cpp

```diff
@@ -497,7 +498,7 @@ namespace azure {  namespace storage_lite {
                     }
 
                     std::istringstream in;
-                    in.rdbuf()->pubsetbuf(buffer, length);
```

Comment:
rdbuf()->pubsetbuf() wont work on mac.

Reply:
I cannot find any documentation showing that `rdbuf()->pubsetbuf()` would behave differently on Mac OS, have you verified the buffer does contain correct information? I think the other possibility is that `ifs.read(buffer, length)` is somehow broken if the file has certain kind of data inside.

Reply:
Tried a simple program on my mac 
#include <iostream>
#include <sstream>
#include <string>
#include <string.h>

using namespace std;

void checkStringstream(std::istream &s, int len)
{
        char buf[101]={0};
        s.read(buf,len);
        cout << "----->Read in the checkStringstream:*" << buf << "*" << endl;

}

int main(void)
{
        std::istringstream s;
        size_t chunk=10;
        char *buffer = (char *) malloc(sizeof(char)* 15);

        memset(buffer,0,15);
        strcpy(buffer,"0123456789");
        cout << "Buffer Contents: " << buffer << endl;

        s.rdbuf()->pubsetbuf(buffer, chunk);

        checkStringstream(s, chunk);
        free(buffer);
        return 0;
}

Buffer Contents: 0123456789
----->Read in the checkStringstream:**

Reply:
`
#include <iostream>
#include <sstream>
#include <string>
#include <string.h>

using namespace std;

void checkStringstream(std::istream &s, int len)
{
        char buf[101]={0};
        s.read(buf,len);
        cout << "----->Read in the checkStringstream:*" << buf << "*" << endl;
}

int main(void)
{
        std::istringstream s;
        size_t chunk=10;
        char *buffer = (char *) malloc(sizeof(char)* 15);

        memset(buffer,0,15);
        strcpy(buffer,"0123456789");
        cout << "Buffer Contents: " << buffer << endl;

        s.rdbuf()->pubsetbuf(buffer, chunk);

        checkStringstream(s, chunk);
        free(buffer);
        return 0;
}
`
`Buffer Contents: 0123456789
----->Read in the checkStringstream:**`

Reply:
`#include <iostream>
#include <sstream>
#include <string>
#include <string.h>

using namespace std;

void checkStringstream(std::istream &s, int len)
{
        char buf[101]={0};
        s.read(buf,len);
        cout << "----->Read in the checkStringstream:*" << buf << "*" << endl;
}

int main(void)
{
        std::istringstream s;
        size_t chunk=10;
        char *buffer = (char *) malloc(sizeof(char)* 15);

        memset(buffer,0,15);
        strcpy(buffer,"0123456789");
        cout << "Buffer Contents: " << buffer << endl;

        s.rdbuf()->pubsetbuf(buffer, chunk);

        checkStringstream(s, chunk);
        free(buffer);
        return 0;
}`

Reply:
``
#include <iostream>
#include <sstream>
#include <string>
#include <string.h>

using namespace std;

void checkStringstream(std::istream &s, int len)
{
        char buf[101]={0};
        s.read(buf,len);
        cout << "----->Read in the checkStringstream:*" << buf << "*" << endl;
}

int main(void)
{
        std::istringstream s;
        size_t chunk=10;
        char *buffer = (char *) malloc(sizeof(char)* 15);

        memset(buffer,0,15);
        strcpy(buffer,"0123456789");
        cout << "Buffer Contents: " << buffer << endl;

        s.rdbuf()->pubsetbuf(buffer, chunk);

        checkStringstream(s, chunk);
        free(buffer);
        return 0;
}
``

Reply:
@katmsft Here is the sample code i ran on my Mac (Ran on 10.12.6 and 10.14.6) both show the same behavior
It's the same behavior on Microsoft Windows too.
 
```
#include <iostream>
#include <sstream>
#include <string>
#include <string.h>
using namespace std;

void checkStringstream(std::istream &s, int len)
{
        char buf[101]={0};
        s.read(buf,len);
        cout << "----->Read in the checkStringstream:*" << buf << "*" << endl;
}
int main(void)
{
        std::istringstream s;
        size_t chunk=10;
        char *buffer = (char *) malloc(sizeof(char)* 15);
        memset(buffer,0,15);
        strcpy(buffer,"0123456789"); 
        cout << "Buffer Contents: " << buffer << endl;
        s.rdbuf()->pubsetbuf(buffer, chunk);
        checkStringstream(s, chunk);
        free(buffer);
        return 0;
}
```

```
macos-10:testprogs jenkinsadmin$ clang++ --std=c++14 -g  teststringstream.cpp
macos-10:testprogs jenkinsadmin$ ./a.out
Buffer Contents: 0123456789
----->Read in the checkStringstream:**
```

Reply:
A simple solution is to call `std::istringstream in(buffer, length)` instead. However this copies the buffer and may cause potential performance drop. You can test it out and see if the problem persist.
So L500  will be  `std::istringstream in(buffer, length)` and L501 should be removed.